nwapp/utils/notifier.js

```diff
@@ -28,14 +40,112 @@ function playNotificationSound() {
   audio = null;
 }
 
-module.exports = function (spec) {
-  if (!settings.showNotifications) return;
-  playNotificationSound();
 
-  var notification = new window.Notification(spec.title, {
-    body: spec.message,
-    icon: getNotificationImageForUri(spec.link)
+// Cache lookup
+var urlFilePathCacheMap = new Map();
+var tempAvatarDirectory = path.join(os.tmpDir(), 'gitter-notification-avatars_');
+log.info('Saving notification avatars at: ' + tempAvatarDirectory);
+var ensureTempAvatarDirectory = ensureDir(tempAvatarDirectory)
+  .then(function() {
+    return tempAvatarDirectory;
   });
+events.on('app:quit', function() {
+  log.info('App is quiting, cleanup avatars');
+  rimraf(tempAvatarDirectory)
```

Comment:
I thought about this as well and just wasn't sure as it seemed to perform fine in basic testing and though the node process would wait until any calls finished. I will update just to be safe.